### DISCUSSION TITLE
th_to_csv converter tool

### DISCUSSION THREAD

**Discussion by dr0n1st** [OpenRadioss/OpenRadioss#746] [Q&A]
**The Problem**
The th_to_csv tool does not work properly.

1. For global variables and parts: it does not convert the velocities.
2. For interfaces: it does not convert the name of the variables, it indicates "var XX"

**Expected behavior**
I expect to 
1. have the possibility to get the velocities for the whole studied system (global variables) and for the sub-systems (parts).
2. identify the outputs by their names, not by guessing.

**Desktop (please complete the following information):**
- Version OpenRadioss 20220930
- CentOS 8

**Comment by sebastienVilleneuve** [comment]
Hi,

Velocity is not a global variable.
Concerning Part , you can output XCG , YCG, ZCG if needed , as far as i know, no velocity is available for th variable per part . ( Please check Radioss doc)

Concerning the second point , I let you check 
[README.md](https://github.com/OpenRadioss/OpenRadioss/tree/main/tools/th_to_csv#readme) :
To have full variable names in .csv file, add /TH/TITLE in 1.rad file when running engine : /TH/TITLE write some _TITLES file that contains additional information allowing to have full titles in writen .csv file
Reason , is that T01 does not contain this information 

Regards,
Sébastien

**Comment by dr0n1st** [comment]
Hi,

Thank you for your quick answer.

I am a Radioss user. I have run the same simulation on both Radios 2022 and OpenRadioss.
I would like to post-process my results with python (which is more convenient for my usage than HyperGraph).
For that, I need to have access to the information that I usually use (cf. attached documents).

I have a .thy file, not a .t01.
I have added /TH/TITLE in the 01.rad file before running OpenRadioss and this is what I get: 

_time,"INTERNAL ENERGY","KINETIC ENERGY","X-MOMENTUM","Y-MOMENTUM","Z-MOMENTUM","MASS","TIME STEP","ROTATION ENERGY","EXTERNAL WORK","SPRING ENERGY","CONTACT ENERGY","HOURGLASS ENERGY","ELASTIC CONTACT ENERGY","FRICTIONAL CONTACT ENERGY","DAMPING CONTACT ENERGY ","C IE ","C KE ","C XMOM ","C YMOM ","C ZMOM ","C MASS ","C HE ","C HEAT ","D IE ","D KE ","D XMOM ","D YMOM ","D ZMOM ","D MASS ","D HE ","D HEAT "," Contact 1 Contact var 32"," Contact 1 Contact var 33"," Contact 1 Contact var 34"," Contact 1 Contact var 35"," Contact 1 Contact var 36"," Contact 1 Contact var 37"," Contact 1 Contact var 38"," Contact 1 Contact var 39"_

![global_var](https://user-images.githubusercontent.com/115089003/194097852-6d4aaa04-9a1f-4afc-93bc-4d6724fc282d.JPG)

![part_var](https://user-images.githubusercontent.com/115089003/194097908-8f657700-6040-4000-8a78-330d85445441.JPG)

![contact_var](https://user-images.githubusercontent.com/115089003/194097946-cb0f2257-c0c9-4ffd-8c30-3224b84b1b64.JPG)

Please, let me know if it is not clear enough.

Best regards,

**Comment by sebastienVilleneuve** [comment]
Hi,

the outputed file contains what necessary to write some names that are used to identify output ( in place of varxxx ) ,
If question is : why X-NORMAL FORCE ( for example)  is not there , it is because this is not embeded in Radioss itself , it is certainly in the post you're using.


Sébastien

**Comment by dr0n1st** [comment]
Hello,

Sorry but I do not understand. What do you mean by "_this is not embeded in Radioss itself , it is certainly in the post you're using_".

The .thy file should contain all the outputs that I requested in HyperMesh, right ? 

About the variables names, the conversion tool should name what are those "var 32" to "var 39".

Best regards,

**Comment by sebastienVilleneuve** [comment]
Hi,

Names are not stored in the T01 file , this file store some indexes that are used by the post to fill the name ( names are stored in the post)

Sébastien
